### PR TITLE
Support vanilla Hawthorn through Open edX plugins

### LINKED ISSUE #3: Support installation on vanilla edx/edx-platform and configuration
This would be a stretch goal, but I believe it can be done. This would seriously help wide adoption in the community.

### PR DESCRIPTION
**Pull request by OmarIthawi**
### Overview

To make the installation steps for Hawthorn+:

`$ pip install figures=x.y.z`
`/edx/bin/edxapp-migrate-lms` :fire: 

### Screenshot

<kbd></kbd>

### Too Long; Don't Read

So, I made a lot of noise about it but finally got the time to actually get it done.

This pull request uses the [Open edX plugins architecture](https://github.com/edx/edx-platform/blob/open-release/ironwood.master/openedx/core/djangoapps/plugins/README.rst) that is supported since Hawthorn.

I dare to say that this also supports Ironwood, but it's early to say without really testing it.

 - [x] ~This PR breaks a lot of tests, and I'm planning to fix them. Let's take on a quick first pass before I fully continue in testing and fixing tests.~ Now all of them are fixed.

Fixes #3

### CODE REVIEW

**Comment by OmarIthawi** [comment]
[(6) Figures: Add Tahoe Hawthorn support for Figures](https://trello.com/c/kLNnDFlP/3598-6-figures-add-tahoe-hawthorn-support-for-figures)

**Comment by lpm0073** [comment]
dude, that's fantastic news. thank you!!!! :D

**Comment by OmarIthawi** [comment]
Thanks for passing by @lpm0073! I'm soo guilty of not making this pull request earlier!

File: figures/pagination.py [hunk 1]
```diff
@@ -1,13 +1,11 @@
-'''Paginatiors for Figures
-
+'''
```
Comment (johnbaldwin) [inline figures/pagination.py]: @OmarIthawi Might as well change to double quotes while you're modifying this

Comment (johnbaldwin) [reply figures/pagination.py]: @OmarIthawi Please see https://www.python.org/dev/peps/pep-0008/#documentation-strings

Comment (OmarIthawi) [reply figures/pagination.py]: :eyes: 

I see PEP-0008, but I don't like what I see.

I'll just keep it as-is!

Comment (OmarIthawi) [reply figures/pagination.py]: Reverted.

**Comment by OmarIthawi** [comment]
@johnbaldwin I've rebased the branch over `hawthorn-upgrade`. I think it's ready to be merged.

One thing I've done, no Ficus support anymore here!

**Comment by OmarIthawi** [comment]
@johnbaldwin any idea how to convince CodeCov that it's ok?

**Comment by johnbaldwin** [comment]
@OmarIthawi I'll look into this today

**Comment by johnbaldwin** [comment]
@OmarIthawi It appears the issue is that code coverage dropped too low. We need test coverage for figures.apps. I need to look into how to unit test apps. I can add more test coverage for figures.sites and figures.permissions since those are pretty weak

**Comment by OmarIthawi** [comment]
Thanks @johnbaldwin! I'm done here, as far as I know. Please merge if it looks good to you.

**Pull request merged**